#!/usr/bin/env python

PACKAGE = "delta"

from dynamic_reconfigure.parameter_generator_catkin import *
from aerial_robot_msgs.msg import DynamicReconfigureLevels

gen = ParameterGenerator()
gen.add("q_x_p_xy", double_t, DynamicReconfigureLevels.RECONFIGURE_Q_X_P_XY, "q_x_p_xy", 300, 0, 500)
gen.add("q_x_p_z",  double_t, DynamicReconfigureLevels.RECONFIGURE_Q_X_P_Z,  "q_x_p_z",  400, 0, 500)
gen.add("q_x_v_xy", double_t, DynamicReconfigureLevels.RECONFIGURE_Q_X_V_XY, "q_x_v_xy", 10,  0, 500)
gen.add("q_x_v_z",  double_t, DynamicReconfigureLevels.RECONFIGURE_Q_X_V_Z,  "q_x_v_z",  10,  0, 500)
gen.add("q_x_a_xy", double_t, DynamicReconfigureLevels.RECONFIGURE_Q_X_A_XY, "q_x_a_xy", 100, 0, 500)
gen.add("q_x_a_z", double_t,  DynamicReconfigureLevels.RECONFIGURE_Q_X_A_Z,  "q_x_a_z",  200, 0, 500)
gen.add("q_x_w_xy", double_t, DynamicReconfigureLevels.RECONFIGURE_Q_X_W_XY, "q_x_w_xy", 5,   0, 500)
gen.add("q_x_w_z",  double_t, DynamicReconfigureLevels.RECONFIGURE_Q_X_W_Z,  "q_x_w_z",  5,   0, 500)
gen.add("r_lambda", double_t, DynamicReconfigureLevels.RECONFIGURE_R_LAMBDA, "r_lambda", 10,  0, 500)

exit(gen.generate(PACKAGE, "delta", "mpc"))
